Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/PortalArtifactSystem.cs

```diff
@@ -0,0 +1,37 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Components;
+using Content.Server.Xenoarchaeology.XenoArtifacts.Events;
+using Content.Shared.Mind;
+using Content.Shared.Storage;
+using Content.Shared.Teleportation.Systems;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Random;
+
+namespace Content.Server.Xenoarchaeology.XenoArtifacts.Effects.Systems;
+
+public sealed class PortalArtifactSystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly LinkedEntitySystem _link = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<PortalArtifactComponent, ArtifactActivatedEvent>(OnActivate);
+    }
+
+    private void OnActivate(Entity<PortalArtifactComponent> artifact, ref ArtifactActivatedEvent args)
+    {
+        var firstPortal = Spawn(artifact.Comp.PortalProto, Transform(artifact.Owner).MapPosition);
+
+        var mindQuery = EntityQuery<MindComponent>().ToList();
```

Comment:
ok really shouldve started review from the start

this is querying mind component *for mind entities*, not MindContainerComponent for the actual mobs

this should also filter where mc.HasMind or something

Reply:
Oh, it unfinished yet, but thanks